### PR TITLE
feat: Support an Impacts All PATH Filter

### PR DESCRIPTION
**Pull request by nikhilbirmiwal**
- Move the sha computation into `prerequisites`. 
- If any paths are added/removed/modified via [dorny's path filter](https://github.com/dorny/paths-filter), skip impacted target computation and upload `ALL`.

Tested using `netkos/act` against a temporary merge graph in staging: https://www.loom.com/share/Library-or-Loom-14-September-2023-4edd8ed7f45240718f4180a3543585dc

Closes https://linear.app/trunk/issue/TRUNK-8676/support-wildcard-impacted-target

### CODE REVIEW

File: .github/workflows/compute_impacted_tests.yaml [hunk 1]
```diff
@@ -10,6 +10,8 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v3
+        with:
```
Comment (nikhilbirmiwal) [inline .github/workflows/compute_impacted_tests.yaml:13]: Clone all. git cloning has been moved to the prerequisites step.

File: .github/workflows/compute_impacted_tests.yaml [hunk 2]
```diff
@@ -31,6 +35,6 @@ jobs:
         shell: bash
         run: |
           target_count=`cat ${{ steps.compute.outputs.impacted_targets_out }} | wc -l`
-          if [[ $target_count -ne 2 ]]; then
+          if [[ $target_count -ne 0 ]]; then
```
Comment (pv72895) [inline .github/workflows/compute_impacted_tests.yaml:38]: Why this change?

**Comment by nikhilbirmiwal** [comment]
Thanks -- FWIW, the PR is queued against an ephemeral trunk-io/merge-action repo, used for testing purposes only.

File: .github/workflows/compute_impacted_tests.yaml (hunk 2 above)
Comment (nikhilbirmiwal) [reply .github/workflows/compute_impacted_tests.yaml:38]: We moved sha computation outside of the compute_impacted_targets script, and expect it as an input into the script. As opposed to running tests with a stable test branch and a moving main, we now run tests with a stable test branch and a stable main, such that we can pin the shas. In that case, we would expect no impacted targets to be computed.

File: action.yaml [hunk 3]
```diff
@@ -29,36 +29,59 @@ inputs:
...
       env:
         DEFAULT_BRANCH: ${{ github.event.repository.default_branch }}
         TARGET_BRANCH: ${{ inputs.target-branch }}
+        PR_BRANCH: ${{ github.head_ref }}
         WORKSPACE_PATH: ${{ inputs.bazel-workspace-path }}
         BAZEL_PATH: ${{ inputs.bazel-path }}
+        IMPACTS_FILTERS_CHANGES: ${{ steps.detect-changed-paths.outputs.changes }}
 
     - name: Install Bazel in PATH
-      if: ${{ steps.prerequisites.outputs.requires_default_bazel_installation == 'true' }}
```
[the remark this answers is not in the document]
Comment (nikhilbirmiwal) [reply action.yaml:47]: I pushed a few changes to cleanup the workflow:
-- always install `jq`, regardless of `impacts_all_detected`
-- set `requires_default_bazel_installation` to false if `impacts_all_detected==true`.

I think this cleans up the workflow, PTAL.
